partiql-tests-data/eval-equiv/spec-tests.ion

```diff
@@ -0,0 +1,582 @@
+            FROM logs AS l
+            WHERE l.co > 1.5
+            GROUP ALL AS g''',
+            '''
+            SELECT COUNT(*) AS largeco
+            FROM logs AS l
+            WHERE l.co > 1.5
+            ''',
+            '''
+            <<
+                {'largeco': COLL_COUNT(SELECT VALUE l
+                                       FROM logs AS l
+                                       WHERE l.co > 1.5
+                                       )
+                }
+            >>''',
+            '''
+            SELECT COUNT(g) AS largeco
+            FROM logs AS l
+            WHERE l.co > 1.5
+            '''
+        ]
+    },
+    {
+        name: "equiv group_all",
+        statement: group_all,
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[{largeco: 0}]
+        }
+    },
+    equiv_class::{
+        id: group_by_with_aggregates,
+        statements: [
+            '''
+            SELECT VALUE {'sensor': sensor,
+                          'avg': COLL_AVG(SELECT VALUE v.l.co FROM g AS v),
+                          'count': COLL_COUNT(g) }
+            FROM logs AS l
+            GROUP BY l.sensor AS sensor GROUP AS g
+            ''',
+            '''
+            SELECT l.sensor AS sensor,
+            AVG(l.co) AS "avg",
+            COUNT(*) AS "count"
+            FROM logs AS l
+            GROUP BY l.sensor
+            '''
+        ]
+    },
+    {
+        name: "equiv group by with aggregates",
+        statement: group_by_with_aggregates,
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[{sensor:1, avg:0.3, count: 2},
+                           {sensor:2, avg:0.3, count: 1}
+            ]
+        }
+    },
+    equiv_class::{
+        id: aliases_from_select_clause,
+        statements: [
+            '''
+            SELECT p.tag || ':' || p.name AS tagname, AVG(p.age) AS average
+            FROM people AS p
+            GROUP BY tagname
+            ''',
+            '''
+            SELECT p.tag || ':' || p.name AS tagname, AVG(p.age) AS average
+            FROM people AS p
+            GROUP BY p.tag || ':' || p.name
+            '''
+        ]
+    },
+    {
+        name: "equiv group by with aggregates",
+        statement: aliases_from_select_clause,
+        env: {
+            people: $bag::[{name: "zoe", age: 10, tag: "child"},
+                           {name: "zoe", age: 20, tag: "adult"},
+                           {name: "bill", age: 30, tag: "adult"}
+            ]
+        },
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[{tagname: "child:zoe", average: 10},
+                {tagname: "adult:zoe", average: 20},
+                {tagname: "adult:bill", average: 30}
+            ]
```

Comment:
Nit formatting. Are we using a standard Ion formatter / stylechecker for this repo?

```suggestion
            people: $bag::[
                {name: "zoe", age: 10, tag: "child"},
                {name: "zoe", age: 20, tag: "adult"},
                {name: "bill", age: 30, tag: "adult"}
            ]
        },
        assert: {
            evalMode: [EvalModeCoerce, EvalModeError],
            result: EvaluationSuccess,
            output: $bag::[
                {tagname: "child:zoe", average: 10},
                {tagname: "adult:zoe", average: 20},
                {tagname: "adult:bill", average: 30}
            ]
```

Reply:
I was following the formatting that the PartiQL spec was using for tests in this PR. I'll investigate further regarding an Ion formatter and update this comment thread. The other `partiql-tests` draft PR (#28), which ported tests from `partiql-lang-kotlin`'s EvaluatorTestSuite.kt, had used the standard `IonTextWriter` pretty printer. I do have some concerns about losing Ion comments when using an Ion formatter.

Reply:
From my own investigating, not sure there's an Ion formatter/linter currently (other than IonJava's IonTextWriter or ion-cli's pretty printer). I'll followup with the Ion team about this next week. I've decided to break up the formatting + linting concerns to a separate issue https://github.com/partiql/partiql-tests/issues/31.